### PR TITLE
Add `interface-over-type-literal` rule

### PR DESCRIPTION
**Pull request by andy-ms**
#### PR checklist

- [X] Addresses an existing issue: #1844
- [X] New feature, bugfix, or enhancement
  - [X] Includes tests
- [X] Documentation update

#### What changes did you make?

Added the `interface-over-type-literal` rule. This forbids `type T = { ... }`, preferring `interface T { ... }`.

### CODE REVIEW

File: src/rules/noEmptyInterfaceRule.ts [hunk 1]
```diff
@@ -27,7 +27,6 @@ export class Rule extends Lint.Rules.AbstractRule {
         rationale: "An empty interface is equivalent to its supertype (or `{}`).",
         optionsDescription: "Not configurable.",
         options: null,
-        optionExamples: ["true"],
```
Comment (andy-ms) [inline src/rules/noEmptyInterfaceRule.ts]: This was left in by accident.

**Comment by nchen63** [comment]
@andy-ms thanks!

**Comment by itajaja** [comment]
Hi, just out of curiosity, can you point me to some rationale behind this?

**Comment by andy-ms** [comment]
It's stylistic. Generally `interface T { foo(): void; }` and `type T = { foo(): void; }` mean the same thing (although an interface can be extended). I think the latter style was mostly coming from people familiar with flow, where there is a difference.

**Comment by itajaja** [comment]
thanks for the explanation :)

**Pull request merged**